### ISSUE TITLE
Custom <pre> / <code> attributes

### ISSUE THREAD

**Issue by naiyerasif** [shikijs/shiki#220]
Please consider a way to add custom attributes on `pre` / `code` elements. Some usecases:

- **Keyboard navigation of codeblocks**: To make the codeblocks accessible for keyboard users, `tabindex = "0"` should be set on `pre` element so that they can scroll them with keyboard. As a short-term goal, we can add `tabindex = "0"` by default. In long-term, we can still keep `tabindex="0"` as default but allow people to customize it. More details can be found here: https://accessibilityinsights.io/info-examples/web/scrollable-region-focusable/
- **Adding [data-*] attributes** for customizing the behavior of codeblocks, e.g., adding a "Copy to Clipboard" button for specific codeblocks, displaying codeblock title, language, etc.

**Comment by fengzilong** [comment]
I'm willing to work on a PR for this feature
